custom/icds_reports/utils/aggregation_helpers/distributed/child_health_monthly.py

```diff
@@ -91,10 +92,13 @@ def _state_aggregation_query(self, state_id):
         alive_in_month = "(child_health.date_death IS NULL OR child_health.date_death - {} >= 0)".format(
             start_month_string
         )
-        migration_status = "(agg_migration.is_migrated=1 AND agg_migration.migration_date < {})::integer".format(
+        migration_status = "(agg_migration.is_migrated = 1 AND agg_migration.migration_date < {})::integer".format(
             start_month_string)
-        seeking_services = "(person_cases.registered_status IS DISTINCT FROM 0 AND {} IS DISTINCT FROM 1)".format(
-            migration_status)
+        registered_status = (
+            "CASE WHEN agg_as.is_registered = 0 AND agg_as.registration_date::date < {start_month_string}"
```

Comment:
Can you explain in plain english what this logic is supposed to do. This logic seems quite odd to me.
This would also be much clearer as a real boolean rather than a case statement with an integer.

Reply:
Can registered status ever be null? This might behave oddly if so

Reply:
Done!

Reply:
This was marked done, but the comment about integers vs booleans was not addressed. Although again, this hard to tell with the rebases